### PR TITLE
core-1646 Add overwrite warning to public tools with ability to allow…

### PR DESCRIPTION
**Pull request by sri2k1us**
… overwrite.

image

### CODE REVIEW

**Review by psarando — CHANGES_REQUESTED** [review]
I'm requesting changes so that we don't attempt to allow non-admins to overwrite public tools (which is not supported in the API).

I also have a suggestion for improving the display of public apps in the confirmation dialog.

File: src/serviceFacades/tools.js [hunk 1]
```diff
@@ -118,17 +118,17 @@ function adminAddTool(tool) {
     });
 }
 
-function updateTool(tool) {
+function updateTool(tool, overwriteAppsAffectedByTool) {
     return callApi({
-        endpoint: `/api/tools/${tool.id}`,
+        endpoint: `/api/tools/${tool.id}?overwrite-public=${overwriteAppsAffectedByTool}`,
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply src/serviceFacades/tools.js:123]: Good catch!

File: src/components/tools/edit/EditTool.js [hunk 2]
```diff
@@ -136,193 +143,269 @@ function EditToolDialog(props) {
...
+                {
+                    appsAffectedByTool?.map((app) => (
+                        <>
+                            <br />
+                            <Typography key={app.id} variant="caption">
+                                * {app.name}
+                            </Typography>
+                        </>
+                    ))
+                }
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply src/components/tools/edit/EditTool.js]: Update to use List:
image

File: src/components/tools/edit/EditTool.js [hunk 3]
```diff
@@ -136,193 +148,273 @@ function EditToolDialog(props) {
...
                 queryClient.invalidateQueries(TOOLS_QUERY_KEY);
                 setUpdateToolError(null);
+                setOverwriteAppsAffectedByTool(false);
+                setAppsAffectedByTool(null);
+                setToolSubmission(null);
                 onClose();
             },
-            onError: setUpdateToolError,
+            onError: (err) => {
+                console.error(getErrorCode(err));
```
Comment (psarando) [inline src/components/tools/edit/EditTool.js:174]: Missed this last review, but this looks like debug logging that can be removed now.

**Review by psarando — APPROVED** [review]
LGTM 👍

**Comment by sri2k1us** [comment]
Thanks for the review.

**Pull request merged**